engine/gameobject/src/gameobject/gameobject_script.cpp

```diff
@@ -557,13 +557,43 @@ namespace dmGameObject
         dmGameObject::HInstance target_instance = dmGameObject::GetInstanceFromIdentifier(dmGameObject::GetCollection(instance), target.m_Path);
         if (target_instance == 0)
             return luaL_error(L, "Could not find any instance with id '%s'.", dmHashReverseSafe64(target.m_Path));
+        dmGameObject::PropertyOptions property_options;
+        property_options.m_Index = 0;
+        bool index_requested = false;
+
+        // Options table
+        if (lua_gettop(L) > 2)
+        {
+            luaL_checktype(L, 3, LUA_TTABLE);
+            lua_pushvalue(L, 3);
+
+            lua_getfield(L, -1, "index");
+            if (!lua_isnumber(L, -1))
+            {
+                return luaL_error(L, "Invalid number passed as index argument in options table.");
+            }
+            // TODO: Check if property is an array and throw error if it isn't
+            property_options.m_Index = luaL_checkinteger(L, -1) - 1;
```

Comment:
What if the index is < 0 (C++)?
Should we return an error?

Reply:
Fixed